kubernetes/istio.go

```diff
@@ -214,26 +215,64 @@ func (in *K8SClient) GetProxyStatus() ([]*ProxyStatus, error) {
 		return nil, errors.New("unable to find any Pilot instances")
 	}
 
+	wg := sync.WaitGroup{}
```

Comment:
Note, I've fetched this PR locally and I guess I missed some review in the past.
This method is always fetching Pods from the wire even if that info is cached.

I think the business layer should combine and pass the pods to this method
.

Reply:
I don't see why we should use the pods from the cache. Using them may lead to scenarios where kiali fetches proxy status from old `istiod` that are either dead or in a not running phase. I think that the most reliable way to perform this operation is to use fresh `istiod` that at this moment are healthy.
What scenario am I missing?

Reply:
Well, it was more to follow same pattern of usage.
But it's an optimization, I understand that when proxy is cached, queries hit the cache for high level info and this fetch to the raw pods happens when building the new proxy status from the istiod status.